frontend/src/metabase/visualizations/components/settings/ChartSettingsListColumns/utils.ts

```diff
@@ -0,0 +1,38 @@
+import type { Column } from "metabase-types/types/Dataset";
+
+import {
+  isAvatarURL,
+  isEntityName,
+  isCategory,
+  isDescription,
+  isDate,
+  isNumeric,
+  isPK,
+} from "metabase-lib/types/utils/isa";
+
+const getIdOrRef = (column: Column | null) => column?.id ?? column?.field_ref;
+
+const findA = (columns: Column[], predicate: (column: Column) => boolean) => {
+  return columns.find(predicate);
+};
+
+export const getDefaultColumns = (columns: Column[]) => {
+  const findInColumns = (predicate: (column: Column) => boolean) =>
+    findA(columns, predicate);
+
+  const imageCols = [
+    findInColumns(isAvatarURL) ?? findInColumns(isEntityName) ?? null,
+  ];
+  const leftCols = [
+    findInColumns(isEntityName) ?? findInColumns(isPK) ?? null,
+    findInColumns(col => isCategory(col) && !isEntityName(col)) ?? null,
+    findInColumns(isDescription) ?? null,
+  ];
+  const rightCols = [findInColumns(isDate) ?? findInColumns(isNumeric) ?? null];
```

Comment:
A pleasure to read!

![giphy](https://user-images.githubusercontent.com/17258145/198284718-43f9d358-af62-47a4-b26b-9cac395c8fb6.gif)